src/v/cloud_storage/partition_recovery_manager.cc

```diff
@@ -364,7 +364,7 @@ partition_downloader::download_log_with_capped_size(
     model::offset_delta start_delta{0};
     for (auto it = offset_map.rbegin(); it != offset_map.rend(); it++) {
         const auto& meta = it->second.meta;
-        if (total_size > max_size) {
+        if (total_size + meta.size_bytes > max_size) {
```

Comment:
This probably needs a special case to download at least one segment.  Otherwise if someone has e.g. 1GB segments, but sets retention to 100MB, then they will recover nothing.

Reply:
Yeah, you are right